src/components/permissions/loginHelpers.ts

```diff
@@ -4,38 +4,37 @@ import { verify, sign } from 'jsonwebtoken'
 import { promisify } from 'util'
 import { PermissionRow, TemplatePermissions, UserInfo } from './types'
 import { compileJWT } from './rowLevelPolicyHelpers'
+import { Organisation } from '../../generated/graphql'
 
 const verifyPromise: any = promisify(verify)
 const signPromise: any = promisify(sign)
 
-const getUsername = async (jwtToken: string) => {
-  let username = 'nonRegistered'
-  if (jwtToken) {
-    try {
-      username = (await verifyPromise(jwtToken, config.jwtSecret)).username
-    } catch (e) {
-      console.log('cannot verify JWT in authorisation header')
-    }
+const getTokenData = async (jwtToken: string) => {
+  try {
+    const data = await verifyPromise(jwtToken, config.jwtSecret)
+    return data
+  } catch (err) {
+    console.log('Cannot parse JWT')
+    throw err
   }
-  return username
 }
 
-const getUserInfo = async (username: string) => {
+const getUserInfo = async (username: string, orgId: number | undefined = undefined) => {
   const templatePermissionRows = await databaseConnect.getUserTemplatePermissions(username)
-  const userInfo = await databaseConnect.getUserDataByUsername(username)
   const {
-    id,
+    userId,
     firstName,
     lastName,
     dateOfBirth,
     email,
   } = await databaseConnect.getUserDataByUsername(username)
+  const organisation = orgId ? await databaseConnect.getOrgById(orgId) : undefined
 
   return {
     username,
     templatePermissions: buildTemplatePermissions(templatePermissionRows),
-    JWT: await getSignedJWT(userInfo, templatePermissionRows),
-    user: { id, firstName, lastName, username, dateOfBirth, email },
+    JWT: await getSignedJWT(username, userId, templatePermissionRows, orgId),
```

Comment:
I think we were planning to not have username in the JWT just the id ?

Reply:
Yeah I left it in because in order to use the same `getUserInfo` method it would need the username. But, as per my last comment, I think I can come up with something better that is also more flexible.